NuRadioMC/EvtGen/NuRadioProposal.py

```diff
@@ -42,35 +42,44 @@
 pp_km = 1.e5
 
 
+def convert_energy_units(pp_energy):
+    """ Convert energy from proposal unit to NuRadioMC unit. """
+    return pp_energy / pp_MeV * units.MeV
+
+
 class SecondaryProperties:
     """
     This class stores the properties from secondary particles that are
     relevant for NuRadioMC, namely:
-    - distance, the distance to the first interaction vertex
-    - energy, the particle energy
-    - shower_type, whether the shower they induce is hadronic or electromagnetic
-    - name, its name according to the particle_name dictionary on this module
+    - distance      : Distance to the first interaction vertex
+    - energy        : Particle energy
+    - shower_type   : Whether the shower they induce is hadronic or electromagnetic
+    - name          : Name according to NuRadioReco.utilities.particle_names
+    - parent_energy : Energy of the parent particle
 
-    Distance and energy are expected to be in NuRadioMC units
+    Distance and energy are in NuRadioMC units
     """
 
     def __init__(self,
                  distance,
                  energy,
                  shower_type,
                  code,
-                 name):
+                 name,
+                 parent_energy):
         self.distance = distance
         self.energy = energy
         self.shower_type = shower_type
         self.code = code
         self.name = name
+        self.parent_energy = parent_energy
 
     def __str__(self):
-        s = "Particle and code: {:} ({:})\n".format(self.name, self.code)
-        s += "Energy in PeV: {:}\n".format(self.energy / units.PeV)
-        s += "Distance from vertex in km: {:}\n".format(self.distance / units.km)
-        s += "Shower type: {:}\n".format(self.shower_type)
+        s =  "Particle and code    : {:} ({:})\n".format(self.name, self.code)
+        s += "Energy               : {:} PeV\n".format(self.energy / units.PeV)
+        s += "Distance from vertex : {:} km\n".format(self.distance / units.km)
+        s += "Shower type          : {:}\n".format(self.shower_type)
+        s += "Parent energy        : {:} PeV".format(self.parent_energy / units.PeV)
```

Comment:
distances/energies maybe easier to read with `{:.2e}` if crossing many energy scales?

Reply:
Its already in units of PeV and km, I think it is okay like this (can be improved later)